lib/traject/config/sirsi_config.rb

```diff
@@ -306,8 +306,8 @@ def extract_sortable_author(author_fields, title_fields, record)
   onexx = Traject::MarcExtractor.cached(author_fields, alternate_script: false, separator: false).collect_matching_lines(record) do |field, spec, extractor|
     non_filing = field.indicator2.to_i
     subfields = extractor.collect_subfields(field, spec).compact
-    subfields[0] = subfields[0].slice(non_filing..-1)
-    subfields.map { |x| x.delete(punct) }.map(&:strip).join(' ')
+    subfields[0] = Array(subfields[0]).slice(non_filing..-1)
```

Comment:
I don't think this is quite right; we're trying to strip some characters from the string, not the list of subfields. I have an alternate fix over in 82e2ca611842cc342c79ce1c6bd72d13a081bf21.